crates/ruff_cli/src/printer.rs

```diff
@@ -389,14 +389,57 @@ fn num_digits(n: usize) -> usize {
         .max(1)
 }
 
-/// Return `true` if the [`Printer`] should indicate that a rule is fixable.
-const fn show_fix_status(autofix_level: flags::FixMode) -> bool {
-    // If we're in application mode, avoid indicating that a rule is fixable.
-    // If the specific violation were truly fixable, it would've been fixed in
-    // this pass! (We're occasionally unable to determine whether a specific
-    // violation is fixable without trying to fix it, so if autofix is not
-    // enabled, we may inadvertently indicate that a rule is fixable.)
-    !autofix_level.is_apply()
+/// Return the number of safe ([`Applicability::Automatic`]) and unsafe ([`Applicability::Suggested`])
+/// fixes generated from a given [`Diagnostics`], contingent upon the mode/applicability levels
+/// passed by the user. Returns [`None`] if there are no applicable fixes.
+fn applicable_fixes(
+    autofix_level: flags::FixMode,
+    diagnostics: &Diagnostics,
+) -> Option<(i32, i32)> {
```

Comment:
I think it would help readability and reusability if you move the logic into its own struct and split into smaller functions (again, naming is hard)

```rust
struct FixableStatistics {
    automatic: u32,
    suggested: u32,
}

impl FixableStatistics {
    fn from_slice(diagnostic: &[Diagnostic]) -> Self {
        // count
    }

    fn has_applicable_fixes(suggested_fixes: SuggestedFixes) -> bool {
        // returns true if there are any applicable fixes
    }
}
```